src/librustc_lint/non_ascii_idents.rs

```diff
@@ -0,0 +1,23 @@
+use crate::lint::{EarlyContext, EarlyLintPass, LintArray, LintContext, LintPass};
+use syntax::ast;
+
+declare_lint! {
+    pub NON_ASCII_IDENTS,
+    Allow,
+    "detects non-ASCII identifiers"
+}
+
+declare_lint_pass!(NonAsciiIdents => [NON_ASCII_IDENTS]);
+
+impl EarlyLintPass for NonAsciiIdents {
+    fn check_ident(&mut self, cx: &EarlyContext<'_>, ident: ast::Ident) {
+        if !ident.name.as_str().is_ascii() {
+            let mut err = cx.struct_span_lint(
```

Comment:
No need to use `let mut err =` here; just chain `.emit()` onto `cx.struct_span_lint(...)`.